solr/core/src/java/org/apache/solr/store/blob/metadata/SharedStoreResolutionUtil.java

```diff
@@ -147,42 +139,53 @@ public static SharedMetadataResolutionResult resolveMetadata(ServerSideMetadata
     
     if (local == null) {
       // The shard index data does not exist locally. All we can do is pull.  
-      // We've computed blobFilesMissingLocally and localFilesMissingOnBlob is empty as it should be.
+      // We've computed blobFilesMissingLocally. localFilesMissingOnBlob is empty as it should be.
       return new SharedMetadataResolutionResult(localFilesMissingOnBlob.values(), blobFilesMissingLocally.values(), blobFilesMissingLocally.values(), false);
     }
 
-    boolean localConflictingWithBlob = false;
+    // If trying to pull files from Blob, make sure similarly named files do not already exist outside the current commit point
+    ImmutableSet<String> allLocalFiles = local.getAllFiles();
+
+    boolean downloadToNewDir = false;
     // Verify there are no inconsistencies between local index and blob index files
     for (BlobFile bf : distant.getBlobFiles()) {
+      final String solrFilename = bf.getSolrFileName();
       // We remove from map of local files those already present remotely since they don't have to be pushed.
-      localFilesMissingOnBlob.remove(bf.getSolrFileName());
-      CoreFileData cf = allLocalFiles.get(bf.getSolrFileName());
-      if (cf != null) {
+      CoreFileData localFile = localFilesMissingOnBlob.remove(solrFilename);
+      if (localFile != null) {
         // The blob file is present locally. Check if there is a conflict between local and distant (blob) versions of that file.
```

Comment:
The blob file is present locally. --> The blob file is present locally and is part of current commit point.